lib/chef/resource/resource_notification.rb

```diff
@@ -20,7 +20,15 @@
 
 class Chef
   class Resource
-    class Notification < Struct.new(:resource, :action, :notifying_resource)
+    class Notification
+
+      attr_accessor :resource, :action, :notifying_resource
+
+      def initialize(resource, action, notifying_resource)
```

Comment:
I think they should be required args.  The only place we use it is https://github.com/chef/chef/blob/master/lib/chef/resource.rb#L1527-L1535.  I know this is an API change but I don't think anyone would be using this class besides us.